### ISSUE TITLE
Model.fit() is not reporting validation accuracy

### ISSUE THREAD

**Issue by SiddGururani** [tflearn/tflearn#699]
I've tried both of the following:
1: Used a floating point argument for validation_set (0.1)
2: Used separate test data for the validation_set.

Either ways, I'm obtaining the following metrics.
```
Training Step: 129  | total loss: 0.00594 | time: 3507.036s
| Adam | epoch: 001 | loss: 0.00594 - acc: 0.9995 -- iter: 0033024/4561552
```
From what I've seen in other issues, model.fit should also report the val_acc and val_loss. May I know what I'm doing incorrectly because of which I'm not seeing these metrics?

**Comment by SiddGururani** [comment]
I feel like the solution to this is that the model is only validated against the validation step whenever I save a snapshot or until an epoch is complete but I'm not sure. Can someone confirm if this is indeed the case?

**Comment by arcyfelix** [comment]
@SiddGururani Yes, I found that myself testing the approach. Sadly, I cannot say why.
